go/vt/vtgate/planbuilder/ddl.go

```diff
@@ -192,64 +192,63 @@ func findTableDestinationAndKeyspace(vschema plancontext.VSchema, ddlStatement s
 	return destination, keyspace, nil
 }
 
-func buildAlterView(ctx context.Context, vschema plancontext.VSchema, ddl *sqlparser.AlterView, reservedVars *sqlparser.ReservedVars, enableOnlineDDL, enableDirectDDL bool) (key.Destination, *vindexes.Keyspace, error) {
-	// For Alter View, we require that the view exist and the select query can be satisfied within the keyspace itself
-	// We should remove the keyspace name from the table name, as the database name in MySQL might be different than the keyspace name
-	destination, keyspace, err := findTableDestinationAndKeyspace(vschema, ddl)
-	if err != nil {
-		return nil, nil, err
-	}
+func buildAlterView(
+	ctx context.Context,
+	vschema plancontext.VSchema,
+	ddl *sqlparser.AlterView,
+	reservedVars *sqlparser.ReservedVars,
+	enableOnlineDDL, enableDirectDDL bool,
+) (key.Destination, *vindexes.Keyspace, error) {
+	return buildCreateViewCommon(ctx, vschema, reservedVars, enableOnlineDDL, enableDirectDDL, ddl.Select, ddl)
+}
 
-	selectPlan, err := createInstructionFor(ctx, sqlparser.String(ddl.Select), ddl.Select, reservedVars, vschema, enableOnlineDDL, enableDirectDDL)
-	if err != nil {
-		return nil, nil, err
-	}
-	selPlanKs := selectPlan.primitive.GetKeyspaceName()
-	if keyspace.Name != selPlanKs {
-		return nil, nil, vterrors.VT12001(ViewDifferentKeyspace)
-	}
-	if vschema.IsViewsEnabled() {
-		if keyspace == nil {
-			return nil, nil, vterrors.VT09005()
-		}
-		return destination, keyspace, nil
-	}
-	isRoutePlan, opCode := tryToGetRoutePlan(selectPlan.primitive)
-	if !isRoutePlan {
-		return nil, nil, vterrors.VT12001(ViewComplex)
-	}
-	if opCode != engine.Unsharded && opCode != engine.EqualUnique && opCode != engine.Scatter {
-		return nil, nil, vterrors.VT12001(ViewComplex)
-	}
-	_ = sqlparser.SafeRewrite(ddl.Select, nil, func(cursor *sqlparser.Cursor) bool {
-		switch tableName := cursor.Node().(type) {
-		case sqlparser.TableName:
-			cursor.Replace(sqlparser.TableName{
-				Name: tableName.Name,
-			})
-		}
-		return true
-	})
-	return destination, keyspace, nil
+func buildCreateView(
+	ctx context.Context,
+	vschema plancontext.VSchema,
+	ddl *sqlparser.CreateView,
+	reservedVars *sqlparser.ReservedVars,
+	enableOnlineDDL, enableDirectDDL bool,
+) (key.Destination, *vindexes.Keyspace, error) {
+	return buildCreateViewCommon(ctx, vschema, reservedVars, enableOnlineDDL, enableDirectDDL, ddl.Select, ddl)
```

Comment:
These two functions look identical now. Can't we rename one and use it for both?